cds_books/literature/covers.py

```diff
@@ -15,11 +15,12 @@
 
 def is_record_with_cover(record):
     """Check if this type of record has cover."""
-    schema = record["$schema"]
-    if schema.endswith("document-v1.0.0.json") or schema.endswith(
-        "series-v1.0.0.json"
-    ):
-        return True
+    if '$schema' in record:
+        schema = record["$schema"]
+        if schema.endswith("document-v1.0.0.json") or schema.endswith(
```

Comment:
question: we are using this `endswith` a lot for matching the right schema comparisons, my concern is that these will fail if we change the schema version.

Reply:
there were discussions on checking the record type instead, although we should decide together about this, and this is only a fix for covers being calculated for intermediate record revision provided by the migration, which does not have `$schema` yet, and it was throwing errors. Also it would happen for any other record as it is a signal in invenio-rest